spec/controllers/lti_deployment_controller_spec.rb

```diff
@@ -187,4 +61,105 @@
       expect(Role.find_by(user: instructor.user, course: Course.find_by(name: 'csc108'))).not_to be_nil
     end
   end
+  describe '#public_jwk' do
+    it 'responds with success when logged out' do
+      get :public_jwk
+      is_expected.to respond_with(:success)
+    end
+    it 'responds with success when logged in' do
+      get_as instructor, :public_jwk
+      is_expected.to respond_with(:success)
+    end
+    it 'responds with a valid jwk' do
+      pub_jwk = get :public_jwk
+      expect { JWT::JWK.new(pub_jwk.body) }.not_to raise_error
```

Comment:
`JWT::JWK.new` will not raise an error as long as its argument is any string. The only thing this test is checking is that the response body is a string. 

Please rewrite this to check that the json response is well formed, the key is in the expected place, that the key is signed with the right secret... I'm sure there are other things we can be checking too

Reply:
I couldn't think of too many additional checks here beyond what you mentioned, but I added a couple of checks for making sure a `kid` param is in the json, that the algorithm is present and correct, and checked to make sure the keys verify a token signed with the key's secret.